### ISSUE TITLE
Feature request: Expose Nunjucks' addGlobal

### ISSUE THREAD

**Issue by robdodson** [11ty/eleventy#495]
I was curious if [addGlobal](https://github.com/mozilla/nunjucks/pull/239) might be something eleventy could expose.

In my instance I wanted to call a function that takes two arguments. It doesn't feel like I could do this correctly with a filter. I could use a shortcode to do this, but then I can't pipe the results into a filter.

Essentially I'm trying to do:

```
{{ findItemInCollection(collection, slug) | dump }}
```

**Comment by Ryuno-Ki** [comment]
Would a syntax like `{{ collection | findItem(slug) | dump }}` work for you?

(I believe I implemented something related to that but haven't packaged it up, yet)

**Comment by robdodson** [comment]
I think that might work but it's sort of arcane syntax. Kind of related to [this comment from the nunjucks maintainer](https://github.com/mozilla/nunjucks/issues/247#issuecomment-96900678) that filters are somewhat awkward syntax on top of function calls.

**Comment by edwardhorsford** [comment]
@robdodson Elventy might not directly support it, but you can add them manually (I discovered this week) by using your own Nunjucks environment. I'm already doing similar to be able to set some specific Nunjucks options.

## Inside your eleventy.js:
```js
  let nunjucksEnvironment = new Nunjucks.Environment(
    new Nunjucks.FileSystemLoader(["src/site/_templates"]), // we need to pass in our includes dir here
    { 
      lstripBlocks: true,
      trimBlocks: true,
      autoescape: false
     });

  nunjucksEnvironment.addGlobal(name, nunjucksGlobals[name]);

  eleventyConfig.setLibrary("njk", nunjucksEnvironment);
```

FWIW I've made a similar filter. I'm pleased to see the syntax matches @Ryuno-Ki's suggested. I also accept an optional arg to do exact matching or not.

I've also just found out that filters (at least for Nunjucks) have access to the context. So I'm tempted to rework my filter to not require collections at all - just take a slug and return the item / array of items / false.

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by zachleat** [comment]
This shipped with #1060 (Eleventy v1)!
